### PR TITLE
Activation scripts

### PR DESCRIPTION
**Pull request by PEZ**
Added behaviour such that if there is an `activate.cljs` in *User scripts* or *Workspace scripts*. directories, then they are run, in the order User, then Workspace.

UPDATE: Now with documentation and an example. Plus some refactoring. A bit unsure about making the uses of the `!app-db` atom spread out, but maybe it is OK? We will not have much state, as I anticipate things, anyway.

### CODE REVIEW

**Comment by PEZ** [comment]
Should `activate.cljs` be placed in the scripts folder, or should we look for them in the respective parent folder? Right now, in this PR, we look for them in _scripts_, meaning that they are listed in the **Run ... script** menus. Maybe that's good?

**Comment by bobisageek** [comment]
My first thought is that I like the idea of being able to re-run my activate script if I want to change something in it, so having it visible feels like a win - it's not treated as 'special', apart from being run automatically.

**Comment by PEZ** [comment]
Thanks, @bobisageek . I'm on that page as well. And then it's up to the user who wants to be able to re-activate to write reloadable activation scripts.

File: src/main/joyride/life_cycle.cljs [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+            ["vscode" :as vscode]
+            [joyride.config :as conf]
+            [joyride.utils :as utils]
+            [promesa.core :as p]))
+
+(def user-init-script "activate.cljs")
+(def user-init-script-path (path/join conf/user-config-path
+                                      conf/user-scripts-path
+                                      user-init-script))
+(def user-init-script-abs-path user-init-script-path)
```
Comment (borkdude) [inline src/main/joyride/life_cycle.cljs:12]: Why is this var the same as the other one?

File: src/main/joyride/life_cycle.cljs [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+                          :script user-init-script
+                          :script-path user-init-script-path
+                          :script-abs-path user-init-script-abs-path}
+                   :workspace {:label "Workspace activate"
+                               :script user-init-script
+                               :script-path workspace-init-script-path
+                               :script-abs-path workspace-init-script-abs-path}})
+
+(defn maybe-run-init-script+ [run-fn {:keys [label script script-path script-abs-path]}]
+  (utils/say (str label " script: " script-path))
```
Comment (borkdude) [inline src/main/joyride/life_cycle.cljs:30]: Should this debugging go away?

File: src/main/joyride/life_cycle.cljs (hunk 1 above)
Comment (PEZ) [reply src/main/joyride/life_cycle.cljs:12]: Yeah, I should just reference the other one twice in the config map. Thanks!

**Comment by borkdude** [comment]
LGTM.

File: src/main/joyride/life_cycle.cljs (hunk 2 above)
Comment (PEZ) [reply src/main/joyride/life_cycle.cljs:30]: It's not debugging. It's progress reporting/logging.

**Comment by PEZ** [comment]
🙏 🎉

**Pull request merged**